contracts/modules/PermissionManager/GeneralPermissionManager.sol

```diff
@@ -8,17 +8,22 @@ import "../Module.sol";
  */
 contract GeneralPermissionManager is IPermissionManager, Module {
 
-    // Mapping used to hold the permissions on the modules provided to delegate
+    // Mapping used to hold the permissions on the modules provided to delegate, delegate add => modules add => permission uint8 => bool 
     mapping (address => mapping (address => mapping (bytes32 => bool))) public perms;
     // Mapping hold the delagate details
     mapping (address => bytes32) public delegateDetails;
+    // Array to track all delegates
+    address[] public allDelegates;
+
+
     // Permission flag
     bytes32 public constant CHANGE_PERMISSION = "CHANGE_PERMISSION";
 
     /// Event emitted after any permission get changed for the delegate
     event ChangePermission(address _delegate, address _module, bytes32 _perm, bool _valid, uint256 _timestamp);
     /// Use to notify when delegate is added in permission manager contract
-    event AddPermission(address _delegate, bytes32 _details, uint256 _timestamp);
+    event LogAddDelegate(address _delegate, bytes32 _details, uint256 _timestamp);
+
```

Comment:
Remove `Log` prefix and mark `_delegate` address as Indexed that may help dApp team